console/console-init/ui/src/modules/address-space/dialogs/CreateMessagingProject/CreateMessagingProject.tsx

```diff
@@ -0,0 +1,211 @@
+import { isMessgaingProjectValid } from "modules/address-space/utils";
+import { ConfiguringEndpoint } from "./ConfiguringEndpoint";
+import { ConfiguringCertificates } from "./ConfiguringCertificates";
+import { ConfiguringRoutes } from "./ConfiguringRoutes";
+
+export interface IMessagingProject {
+  namespace: string;
+  name: string;
+  type?: string;
+  plan?: string;
+  authService?: string;
+  customizeEndpoint: boolean;
+  isNameValid: boolean;
+  certValue?: string;
+  privateKey?: string;
+}
+
+interface ICreateMessagingProjectProps {}
+
+const initialMessageProject: IMessagingProject = {
+  name: "",
+  namespace: "",
+  type: "",
+  isNameValid: true,
+  customizeEndpoint: false
+};
+
+export interface IWizardSteps {
+  showCustomize: boolean;
+  showCertificate: boolean;
+  showRoutes: boolean;
+}
+
+const CreateMessagingProject: React.FunctionComponent<ICreateMessagingProjectProps> = () => {
+  const [messagingProject, setMessagingProject] = useState<IMessagingProject>(
+    initialMessageProject
+  );
+  const [stepStatus, setStepStatus] = useState<IWizardSteps>({
+    showCertificate: false,
+    showCustomize: false,
+    showRoutes: true
+  });
+
+  const { state, dispatch } = useStoreContext();
+  const { modalProps } = (state && state.modal) || {};
+  const { onConfirm, onClose } = modalProps || {};
+
+  const resetFormState = () => {
+    setMessagingProject(initialMessageProject);
+  };
+
+  const refetchQueries: string[] = ["all_address_spaces"];
+  const [setQueryVariables] = useMutationQuery(
+    CREATE_ADDRESS_SPACE,
+    refetchQueries,
+    resetFormState,
+    resetFormState
+  );
+
+  const onCloseDialog = () => {
+    dispatch({ type: types.HIDE_MODAL });
+    if (onClose) {
+      onClose();
+    }
+  };
+
+  const handleSave = async () => {
+    const { name, namespace, type, plan, authService } = messagingProject;
+    if (isMessgaingProjectValid(messagingProject)) {
+      const variables = {
+        as: {
+          metadata: {
+            name: name,
+            namespace: namespace
+          },
+          spec: {
+            type: type?.toLowerCase(),
+            plan: plan?.toLowerCase(),
+            authenticationService: {
+              name: authService
+            }
+          }
+        }
+      };
+      await setQueryVariables(variables);
+
+      onCloseDialog();
+      if (onConfirm) {
+        onConfirm();
+      }
+    }
+  };
+
+  const getInitialStep = () => {
+    const step = {
+      name: "Configuration",
+      component: (
+        <MessagingProjectConfiguration
+          projectDetail={messagingProject}
+          setProjectDetail={setMessagingProject}
+          stepStatus={stepStatus}
+          setStepStatus={setStepStatus}
+        />
+      ),
+      enableNext: isMessgaingProjectValid(messagingProject)
+    };
+    return step;
+  };
+
+  const getConfigurationStep = () => {
+    const getConfiguringStep = () => {
```

Comment:
Is there any specific reason for creating inner functions.
can you please try to use object instead inner function if not causing any issue.

  const getConfigurationStep = () => {    
      const configuringStep = {
        name: "Configuring",
        component: (
          <ConfiguringEndpoint
            projectDetail={messagingProject}
            setProjectDetail={setMessagingProject}
            stepStatus={stepStatus}
            setStepStatus={setStepStatus}
          />
        ),
        enableNext: isMessgaingProjectValid(messagingProject)
      };
      
      const certificatesStep = {
        name: "Certificates",
        component: (
          <ConfiguringCertificates
            projectDetail={messagingProject}
            setProjectDetail={setMessagingProject}
          />
        ),
        enableNext: isMessgaingProjectValid(messagingProject)
      };
  
      const routesStep = {
        name: "Routes",
        component: (
          <ConfiguringRoutes
            projectDetail={messagingProject}
            setProjectDetail={setMessagingProject}
          />
        ),
        enableNext: isMessgaingProjectValid(messagingProject)
      };

    const step = {
      name: "Endpoint customization",
      steps: [
        ...[configuringStep],
        ...(stepStatus.showCertificate ? [certificatesStep] : []),
        ...(stepStatus.showRoutes ? [routesStep] : [])
      ],
      enableNext: isMessgaingProjectValid(messagingProject),
      canJumpTo: isMessgaingProjectValid(messagingProject)
    };
    return step;
  }